ballerina/xmldata.bal

```diff
@@ -38,10 +40,128 @@ public type JsonOptions record {
 # + jsonValue - The JSON source to be converted to XML
 # + options - The `xmldata:xmldata:JsonOptions` record for JSON to XML conversion properties
 # + return - XML representation of the given JSON if the JSON is
-#            successfully converted or else an `xmldata:Error`
-public isolated function fromJson(json jsonValue, JsonOptions options = {}) returns xml?|Error = @java:Method {
-    'class: "io.ballerina.stdlib.xmldata.JsonToXml"
-} external;
+# successfully converted or else an `xmldata:Error`
+public isolated function fromJson(json jsonValue, JsonOptions options = {}) returns xml?|Error {
+    if !isSingleNode(jsonValue) {
+        return getElement("root", check traverseNode(jsonValue, {}, options), check getAttributesMap(jsonValue, options = options));
+    } else {
+        map<json>|error jMap = jsonValue.ensureType();
+        if jMap is map<json> {
+            if jMap.length() == 0 {
+                return xml ``;
+            }
+            return getElement(jMap.keys()[0], check traverseNode(jMap.toArray()[0], {}, options));
+        }
+    }
+    return error Error("failed to parse xml");
+}
+
+isolated function traverseNode(json jNode, map<string> parentNamespaces, JsonOptions options = {}) returns xml|Error {
+    string arrayEntryTag = options.arrayEntryTag == "" ? "item" : options.arrayEntryTag;
+    string attributePrefix = options.attributePrefix == "" ? "@" : options.attributePrefix;
```

Comment:
Ideally, we should prevent passing empty string values and just use the value without this check. need to write compiler plugin for it

Reply:
Let's do it later